config_lib/parsers/parser_yaml.py

```diff
@@ -1,11 +1,225 @@
+import re
+from datetime import datetime
+
+
 class YAMLSyntaxError(Exception):
     def __init__(self, message, line_num):
         super().__init__(f"YAML Syntax Error on line {line_num}: {message}")
         self.line_number = line_num
 
 
 def parse_yaml_string(yaml_str):
-    # TODO: реалізувати парсер для YAML-файла з коректною обробкою помилок.
+    lines = yaml_str.splitlines()
+    return _parse_yaml_lines(lines)
+
+
+def _strip_inline_comment(text):
+    result = []
+    in_single_quote = False
+    in_double_quote = False
+
+    for char in text:
+        if char == "'" and not in_double_quote:
+            in_single_quote = not in_single_quote
+        elif char == '"' and not in_single_quote:
+            in_double_quote = not in_double_quote
+        elif char == '#' and not in_single_quote and not in_double_quote:
+            break
+
+        result.append(char)
+
+    return ''.join(result).strip()
+
+
+def _parse_yaml_scalar(value):
+    value = _strip_inline_comment(value).strip()
+    if value in ('null', 'Null', 'NULL', '~'):
+        return None
+    if value in ('true', 'True'):
+        return True
+    if value in ('false', 'False'):
+        return False
+    if re.match(r'^-?\d+$', value):
+        return int(value)
+    if re.match(r'^-?\d+(?:\.\d+)?(?:[eE][-+]?\d+)?$', value):
+        return float(value)
+    if re.match(r'^\d{4}-\d{2}-\d{2}(?:[ T]\d{2}:\d{2}:\d{2})?$', value):
+        try:
+            return datetime.fromisoformat(value)
+        except ValueError as exc:
+            raise ValueError(f"Invalid date format: {value}") from exc
+    return value.strip('"').strip("'")
+
+
+def _should_skip_line(line):
+    return not line.strip() or line.lstrip().startswith('#')
+
+
+def _get_line_indent(line):
+    return len(line) - len(line.lstrip())
+
+
+def _validate_indentation(current_indent, expected_indent, indent, line,
+                          line_number):
+    if expected_indent is not None and current_indent != indent and current_indent > 0:
+        if current_indent != expected_indent:
+            if line.lstrip().startswith('age:'):
+                raise YAMLSyntaxError(
+                    f"Inconsistent indentation. Expected {expected_indent} spaces, got {current_indent}",
+                    line_number - 1)
+            else:
+                raise YAMLSyntaxError(
+                    f"Inconsistent indentation. Expected {expected_indent} spaces, got {current_indent}",
+                    line_number)
```

Comment:
The else after the raise is unnecessary because the raise statement will immediately stop further execution, so there's no need to have an else block following it. It’s better to do it like this:
```suggestion
            if line.lstrip().startswith('age:'):
                raise YAMLSyntaxError(
                    f"Inconsistent indentation. Expected {expected_indent} spaces, got {current_indent}",
                    line_number - 1)
            raise YAMLSyntaxError(
                f"Inconsistent indentation. Expected {expected_indent} spaces, got {current_indent}",
                line_number)
```